plugins/svgtools/include/detray/plugins/svgtools/illustrator.hpp

```diff
@@ -0,0 +1,176 @@
+#include "detray/plugins/svgtools/conversion/surface.hpp"
+#include "detray/plugins/svgtools/conversion/volume.hpp"
+#include "detray/plugins/svgtools/styling/styling.hpp"
+#include "detray/plugins/svgtools/utils/volume_utils.hpp"
+
+// Actsvg include(s)
+#include "actsvg/meta.hpp"
+
+// System include(s)
+#include <array>
+#include <vector>
+
+namespace detray::svgtools {
+
+/// @brief SVG generator for a detector and related entities.
+template <typename detector_t>
+class illustrator {
+
+    public:
+    illustrator() = delete;
+
+    illustrator(const detector_t& detector,
+                const typename detector_t::name_map& name_map)
+        : _detector{detector}, _name_map{name_map} {}
+
+    illustrator(const detector_t& detector,
+                const typename detector_t::name_map& name_map,
+                const styling::style& style)
+        : _detector{detector}, _name_map{name_map}, _style{style} {}
+
+    /// @brief Converts a detray surface in the detector to an actsvg svg.
+    /// @param identification the id of the svg object.
+    /// @param context the geometry context.
+    /// @param detector the detector.
+    /// @param index the index of the surface in the detector.
+    /// @param view the display view.
+    /// @returns SVG of the detector's surface.
+    template <typename view_t>
+    inline auto draw_surface(
+        const std::string& identification,
+        const typename detector_t::geometry_context& context,
+        const size_t index, const view_t& view) const {
+        const auto surface = detray::surface{
+            _detector,
+            _detector.surface_lookup()[static_cast<detray::dindex>(index)]};
+        if (surface.is_portal()) {
+            auto p_portal = svgtools::conversion::portal<point3_container>(
+                context, _detector, surface);
+            svgtools::styling::apply_style(p_portal, _style);
+            return actsvg::display::portal(identification, p_portal, view);
+        }
+        auto p_surface =
+            svgtools::conversion::surface<point3_container>(context, surface);
+        svgtools::styling::apply_style(p_surface, _style);
+        return actsvg::display::surface(identification, p_surface, view);
+    }
+
+    /// @brief Converts a collection of detray surfaces in the detector to an
+    /// actsvg svg.
+    /// @param identification the id of the svg object.
+    /// @param context the geometry context.
+    /// @param detector the detector.
+    /// @param indices the collection of surface indices in the detector to
+    /// convert.
+    /// @param view the display view.
+    /// @returns SVG of the detector's surfaces.
+    template <typename iterator_t, typename view_t>
+    inline auto draw_surfaces(
+        const std::string& identification,
+        const typename detector_t::geometry_context& context,
+        const iterator_t& indices, const view_t& view) const {
+        actsvg::svg::object ret;
+        ret._tag = "g";
```

Comment:
What does this tag mean?

Reply:
It means the svg object is a group (all the surface svg objects are added to the group)

Reply:
Thanks for clarifying :)